packages/zod-openapi/src/lib/zod-openapi.ts

```diff
@@ -201,7 +201,7 @@ function parseObject({
   // `catchall` obviates `strict`, `strip`, and `passthrough`
   if (
     !(
-      zodRef._def.catchall instanceof z.ZodNever ||
+      // zodRef._def.catchall instanceof z.ZodNever ||
```

Comment:
This assuredly misses some things, how could we check this without using instanceof?

Reply:
I actually don't think this misses things, if X is instanceof Y, it will definitely have the same typename.
Of course, things could theoretically have the same typename but not be the same zod type, but looking at zod this seems unlikely to happen.